packages/wrapped-keys/src/lib/lit-actions-client/code-repository.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  LitActionCodeRepository,
+  LitActionCodeRepositoryInput,
+  LitActionType,
+} from './types';
+import { Network } from '../types';
+
+/**
+ * A repository for managing Lit Actions related to blockchain operations.
+ * The repository provides functionalities for handling transactions, messages,
+ * encrypted keys, and private keys across multiple blockchain platforms.
+ * @type {LitActionCodeRepository}
+ */
+const litActionCodeRepository: LitActionCodeRepository = Object.freeze({
+  signTransaction: Object.seal({
+    evm: '',
+    solana: '',
+  }),
+  signMessage: Object.seal({
+    evm: '',
+    solana: '',
+  }),
+  generateEncryptedKey: Object.seal({
+    evm: '',
+    solana: '',
+  }),
+  exportPrivateKey: Object.seal({
+    evm: '',
+    solana: '',
+  }),
+});
+
+/**
+ * Updates the litActionCodeRepository with the provided entries.
+ * @param { LitActionCodeRepositoryInput } repository - user provided repository to set
+ */
+function setLitActionsCode(repository: LitActionCodeRepositoryInput) {
```

Comment:
I tested the behaviour of Object.seal(), and in my browser it _silently fails_ if I try to do something like set `litActionCodeRepository.signTransaction.blah = 'meow'`; no error is thrown to the user -- and this is how `seal()` is documented to work according to the specification.
If someone passes us an invalid key for either the LitActionType or Network, we want an explicit error to be thrown to the caller immediately, since otherwise the silent failure will confuse them (when they call the lit action, it'll execute our default action using CID instead of what they gave us, or whatever action they _last set in the dictionary for the key_.

We should use some explicit typescript assertion functions that throw an informative error to the user if they provide an invalid `LitActionType` or `Network` string instead of using `as` in this context to coerce it without verifying it at runtime.   I wrote these assertions in my PR but I think they got lost somewhere along the way -- see:

https://github.com/LIT-Protocol/js-sdk/blob/135373acdc013becf0708a2898ae40652e616ab3/packages/wrapped-keys/src/lib/lit-actions-client/code-repository.ts#L28-L79

Reply:
I removed those intentionally as types are already validating that users don't pass extra actions nor networks. But you are right, in a JS environment like browser that won't be enough and it will confuse consumers

Also added tests 🙂